src/core/batch.rkt

```diff
@@ -0,0 +1,101 @@
+#lang racket
+
+(require "../utils/timeline.rkt")
+
+(provide progs->batch
+         batch->progs
+         (struct-out batch)
+         get-expr)
+
+(struct batch (nodes roots vars nodes-length))
+
+(define (progs->batch exprs #:timeline-push [timeline-push #f] #:vars [vars '()] #:taylor [taylor #f])
+  (define icache (reverse vars))
+  (define exprhash
+    (make-hash (for/list ([var vars] [i (in-naturals)])
+                 (cons var i))))
+  ; Counts
+  (define size 0)
+  (define exprc 0)
+  (define varc (length vars))
+
+  ; Translates programs into an instruction sequence of operations
+  (define (munge prog)
+    (set! size (+ 1 size))
+    (define node ; This compiles to the register machine
+      (match prog
+        [(list '- arg1 arg2)
+         #:when taylor
```

Comment:
instead of using `#:taylor` on every line, it can go to a separate function

Reply:
Oof, yeah, this is pretty ugly. I'd probably rather it be a separate function, even if that's slow. It would be nice if we could do this as a `batch->batch` function. Probably doable? You'd need to store a vector on the side to update references.

Reply:
I could also imagine doing this purely through egg if we could make egg really fast. But for now let's just do this as a separate `batch->batch` function called, like, `expand-taylor` or something. We could make that a separate PR, just to get stuff merged.